plugins/multilanguage.ts

```diff
@@ -176,7 +183,56 @@ export default function multilanguage(userOptions: Options) {
           document.head.appendChild(meta);
           document.head.appendChild(document.createTextNode("\n"));
         }
+
+        if (page.data.alternateX) {
+          addHreflangX(page.data.alternateX, site, document);
+        }
       }
     });
   };
 }
+
+function getAlternateXData(
+  currentPage: Page<Data>,
+  validList: string[],
+  filteredPages: Page<Data>[],
+): Data | undefined {
+  const langX = currentPage.data.langX;
+
+  if (!langX) return void 0;
+
+  // Use case when langX is URL string
+  if (langX.startsWith("/")) {
+    const regUrl = new RegExp(`^\/${trim(langX, "\/")}\/?`, "i");
+    return filteredPages.filter(({ data }) => {
+      data.basename;
+      const [isMatch] = data.url.match(regUrl) ?? [];
+      return !!isMatch;
+    })
+      ?.[0]?.data;
```

Comment:
This doesn't work in all cases because the basename is not unique (two pages in different folders can have the same basename).
The correct way to get this is by checking the entire url:

```js
const pageX = filteredPages.find((page) => page.data.url === langX);
```

Reply:
If no page was found, I would warn it (`log.warn(...)`).

Reply:
> the basename is not unique

My bad, I put it there because I was consider the `basename` and found out it unecessary to think about it. Really amatuer for not remove that line.

>```ts
>const pageX = filteredPages.find((page) => page.data.url === langX);
>```

This actually was my first solution, however, I have to use regex as the `data.url` can have the trailing `/`.

Reply:
> If no page was found, I would warn it (`log.warn(...)`).

Good idea, I've updated the warning log:

![image](https://github.com/lumeland/lume/assets/68631458/9dc07d07-0349-4527-b022-60fec39398fe)

How do you think?

Reply:
Nice!